src/core/Toggle.tsx

```diff
@@ -4,22 +4,38 @@ import cn from "./utils/cn";
 
 type ToggleProps = {
   id: string;
+  size?: "sm" | "lg";
   label?: string;
   className?: string;
 } & React.ButtonHTMLAttributes<HTMLButtonElement>;
 
-const Toggle: React.FC<ToggleProps> = ({ id, label, className, ...props }) => {
+const Toggle: React.FC<ToggleProps> = ({
+  id,
+  size = "lg",
+  label,
+  className,
+  ...props
+}) => {
+  const rootSize = size === "sm" ? "w-[42px] h-[24px]" : "w-[56px] h-32";
+  const thumbSize =
+    size === "sm"
+      ? "w-[21px] h-[21px] translate-x-1 data-[state=checked]:translate-x-[20px]"
+      : "h-[28px] w-[28px] translate-x-2  data-[state=checked]:translate-x-[26px]";
+
   return (
     <div className="flex items-center">
       <Switch.Root
         className={cn(
-          "p-0 h-32 w-[56px] bg-neutral-600 rounded-full relative inline-block transition-colors data-[disabled]:bg-gui-unavailable data-[disabled]:cursor-not-allowed data-[state=checked]:bg-orange-600 focus:outline-none focus-visible:outline-offset-0 focus-visible:outline-4 focus-visible:outline-gui-focus",
+          "p-0 bg-neutral-600 rounded-full relative inline-block transition-colors data-[disabled]:bg-gui-unavailable data-[disabled]:cursor-not-allowed data-[state=checked]:bg-orange-600 focus:outline-none focus-visible:outline-offset-0 focus-visible:outline-4 focus-visible:outline-gui-focus",
```

Comment:
Heads up there may be a conflict here with that `focus-base` class I added in the Badge PR (which would remove these `focus`/`focus-visible` classes), but we can resolve that when it comes to it